next permutation in c++

```diff
@@ -0,0 +1,69 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+//to print a vector
+void print(vector<int> a,int n){ 
+	for(int i=0;i<n;i++)
+		cout<<a[i]<<" ";
+	cout<<endl;
+}
+
+void nextPermutation(vector<int> a,int n){
+	int k=-1,l,temp;
+	
+	//finding largest k s.t. a[k]<a[k+1]
+	for(int i=0;i<n-1;i++){
+		if(a[i]<a[i+1])
+			k=i;
+	}
```

Comment:
The C++17 standard library has `std::max_element()` and `std::min_element()` functions, which can help here: https://en.cppreference.com/w/cpp/algorithm/max_element